src/openrct2/peep/Guest.cpp

```diff
@@ -5103,6 +5103,55 @@ void Guest::UpdateRide()
 static void peep_update_walking_break_scenery(Peep* peep);
 static bool peep_find_ride_to_look_at(Peep* peep, uint8_t edge, uint8_t* rideToView, uint8_t* rideSeatToView);
 
+/**
+ *
+ *  rct2: 0x699F5A
+ * al:thoughtType
+ * ah:thoughtArguments
+ * esi: peep
+ */
+void Peep::InsertNewThought(PeepThoughtType thoughtType, uint8_t thoughtArguments)
+{
+    PeepActionType action = PeepThoughtToActionMap[thoughtType].action;
+    if (action != PEEP_ACTION_NONE_2 && this->action >= PEEP_ACTION_NONE_1)
+    {
+        this->action = action;
+        this->action_frame = 0;
+        this->action_sprite_image_offset = 0;
+        this->UpdateCurrentActionSpriteType();
+        this->Invalidate();
```

Comment:
Again all of these `this->` aren't required. Some people do like this style but this codebase does not do it like this.